packages/components/src/internal/components/assay/SpecialtyAssayPanel.tsx

```diff
@@ -61,15 +72,44 @@ export const SpecialtyAssayPanel: FC<SpecialtyAssayPanelProps> = memo(props => {
                     )}
                 </Col>
             </Row>
-            <Row>
-                <Col xs={6}>
-                    <div className={warning ? 'margin-bottom' : 'margin-top margin-bottom'}>
-                        <b>Supported File Types</b>
-                    </div>
-                    <p>{selected?.fileTypes.join(', ')}</p>
-                </Col>
-            </Row>
+            {selected && options?.length > 0 &&
+                <Row>
+                    <Col xs={6}>
+                        <div className={warning ? 'margin-bottom' : 'margin-top margin-bottom'}>
+                            <b>Supported File Types</b>
+                        </div>
+                        <p>{selected?.fileTypes.join(', ')}</p>
+                    </Col>
+                </Row>
+            }
             {children}
+            {!hasPremium && (
```

Comment:
I haven't looked at the LK module implementations yet, but is there a reason for passing in this `hasPremium` property instead of just having this component check for the "has premium module" itself? I think we have a few other checks like that in the ui-components already.

Reply:
Yeah I was just going for the dumb component.  Wasn't sure of all the use cases particularly in our apps, where we may never want to show the upsell message. Also easier to jest test and show in storybook as a prop.